src/main/java/org/tomitribe/auth/signatures/Verifier.java

```diff
@@ -83,14 +83,20 @@ public Verifier(final Key key, final Signature signature, final Provider provide
     }
 
     public boolean verify(final String method, final String uri, final Map<String, String> headers) throws IOException, NoSuchAlgorithmException, SignatureException {
-
+        try {
+            signature.verifySignatureValidityDates();
+        } catch (Exception e) {
+            return false;
```

Comment:
what happens here? swallowing exception?
Can we issue a small debug log statement?

Reply:
The `verify` method returns a boolean, and initially I didn't pay attention to the fact that it can also throw an exception. Let me change that to throwing an exception, that provides more information about why the verification is failing.

Reply:
The code is now throwing an exception.